sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/SharedAccessSignatureCredential.cs

```diff
@@ -84,5 +84,17 @@ public SharedAccessSignatureCredential(SharedAccessSignature signature)
         ///
         public override ValueTask<AccessToken> GetTokenAsync(TokenRequestContext requestContext,
                                                              CancellationToken cancellationToken) => new ValueTask<AccessToken>(GetToken(requestContext, cancellationToken));
+
+        internal void UpdateSharedAccessKey(string keyName, string keyValue)
+        {
+            lock (ExtensionSyncRoot)
```

Comment:
nit:  Now that we're using it for multiple purposes, we should probably think about giving the lock object a more generic name.   Maybe something like "SignatureSyncRoot" or similar?